main.rb

```diff
@@ -0,0 +1,97 @@
+require_relative 'app'
+require_relative 'nameable'
+require_relative 'book'
+require_relative 'rental'
+require_relative 'student'
+require_relative 'teacher'
+
+def main
+  books = [], people = [], rentals = []
+  menu_options = [
+    { number: 1, label: 'List all books', action: -> { list_books(books) } },
+    { number: 2, label: 'List all people', action: -> { list_people(people) } },
+    { number: 3, label: 'Create a person', action: -> { create_person_option(people) } },
+    { number: 4, label: 'Create a book', action: -> { create_book_option(books) } },
+    { number: 5, label: 'Create a rental', action: -> { create_rental_option(books, people, rentals) } },
+    { number: 6, label: 'List rentals for a person', action: -> { list_rentals_option(people, rentals) } },
+    { number: 7, label: 'Quit', action: -> { exit_app } }
+  ]
+
+  loop do
+    display_menu(menu_options)
+    choice = gets.chomp.to_i
+    if choice.between?(1, menu_options.size)
+      menu_options[choice - 1][:action].call
+    else
+      invalid_choice
+    end
+    break if choice == 7
+  end
+end
+
+def display_menu(menu_options)
+  puts 'Options:'
+  menu_options.each { |option| puts "#{option[:number]}. #{option[:label]}" }
+  print 'Enter your choice: '
+end
+
+def menu_options
+  [
+    { number: 1, label: 'List all books', action: method(:list_books) },
+    { number: 2, label: 'List all people', action: method(:list_people) },
+    { number: 3, label: 'Create a person', action: method(:create_person_option) },
+    { number: 4, label: 'Create a book', action: method(:create_book_option) },
+    { number: 5, label: 'Create a rental', action: method(:create_rental_option) },
+    { number: 6, label: 'List rentals for a person', action: method(:list_rentals_option) },
+    { number: 7, label: 'Quit', action: method(:exit_app) }
+  ]
+end
+
+def create_person_option(people)
+  print 'Enter person name: '
+  name = gets.chomp
+  print 'Enter person type (teacher or student): '
+  type = gets.chomp.downcase
+  print 'Enter ID for person: '
+  id = gets.chomp.to_i
+  person = create_person(id, name, type)
+  people << person if person
+end
```

Comment:
- [ ] [REQUIRED] Kindly ensure that you replicate the design as in the video guideline provided when a user tries to create a person.  You skipped the age of the person, you also forgot to add options 1 and 2 to the student and teacher, and you also forgot to ask for parent permission for the student option. Let's implement these recommendations before asking for another review.
### Your Design
<img width="284" alt="Screenshot 2023-09-15 at 16 07 36" src="https://github.com/najibullahjafari/oop-school-library/assets/58692590/a0e903bd-8c48-4583-8396-e248fc29a098">

### The guideline provided
<img width="510" alt="Screenshot 2023-09-15 at 16 05 36" src="https://github.com/najibullahjafari/oop-school-library/assets/58692590/6ba93b25-40fa-4af0-9df5-95eea4dc1424">